pkg/dr/collection.go

```diff
@@ -104,70 +104,89 @@ type DRCollection struct {
 	NumberOfFiles      int64
 }
 
+type ouPath struct {
+	ou   string
+	path string
+}
+
 func GetAllCollections(config Config) (chan *DRCollection, error) {
-	acc, err := NewAccount(config)
+
+	accounts, err := NewServiceAccounts(config)
 	if err != nil {
 		return nil, err
 	}
 
-	cpaths := make(chan string, 10000)
+	cpaths := make(chan ouPath, 10000)
 	go func() {
 
 		defer close(cpaths)
 
-		conn := connection.NewIRODSConnection(acc, 300*time.Second, "dr-gateway")
-		if err := conn.Connect(); err != nil {
-			log.Errorf("connection failure: %s", err)
-			return
-		}
-		defer conn.Disconnect()
-
 		for _, ou := range config.OrganisationalUnits {
-			if _colls, err := fs.SearchCollectionsByMeta(conn, "organisationalUnit", ou); err != nil {
+
+			acc := accounts[ou.Name]
+
+			conn := connection.NewIRODSConnection(acc, 300*time.Second, "dr-gateway")
+			if err := conn.Connect(); err != nil {
+				log.Errorf("connection failure: %s", err)
+				return
+			}
+			defer conn.Disconnect()
+
+			if _colls, err := fs.SearchCollectionsByMeta(conn, "organisationalUnit", ou.Name); err != nil {
 				log.Errorf("%s\n", err.Error())
 			} else {
 				for _, c := range _colls {
-					cpaths <- c.Path
+					cpaths <- ouPath{
+						ou.Name,
+						c.Path,
+					}
 				}
 			}
 		}
 	}()
 
-	return getCollections(acc, cpaths), nil
+	return getCollections(accounts, cpaths), nil
 }
 
 func FindCollectionsByMeta(config Config, key, value string) (chan *DRCollection, error) {
```

Comment:
This function can be reused in `GetAllCollections`

Reply:
It's not trivial to reuse it given the internal dependency of OU.